Generalizing build script

jewzaam opened the pull request:
- switch to make
- templatized manifests
- more robust catalog generation
- many fixes from review of original PR: https://github.com/rogbas/osd-operators/pull/1

jewzaam committed:
Generalizing build script

- switch to make
- templatized manifests
- more robust catalog generation
- many fixes from review of original PR: https://github.com/rogbas/osd-operators/pull/1

jewzaam committed:
Added OWNERS

rogbas commented on Makefile:
```diff
@@ -0,0 +1,153 @@
+	popd
+
+.PHONY: operator-source
+operator-source: $(SOURCE_DIR)/dedicated-admin-operator
```
dedicated-admin is hardcoded, shouldn't this be a loop thru a list as well?

jewzaam replied:
I added an update where it's passed to a function.  We could make it a list somewhere in a config file, but I prefer it being referenced in the `Makefile`.  It's less to reason about to understand how it works.

rogbas commented on Makefile:
```diff
@@ -0,0 +1,153 @@
+CURRENT_COMMIT=$(shell git rev-parse --short=8 HEAD)
+CATALOG_VERSION?=$(CHANNEL)-$(BUILD_DATE)-$(CURRENT_COMMIT)
+
+SUBSCRIPTIONS=$(shell cat subscriptions.json)
```
This seems to be obsolete, file does not exist and variable is not being used.

rogbas commented on catalog-manifests/dedicated-admin-operator/dedicated-admin-operator.package.yaml:
```diff
@@ -0,0 +1,4 @@
+channels:
+- currentCSV: dedicated-admin-operator.v0.1.67-2019-04-09-091172db
+  name: generalize-builds
```
I imagine this channel name should be renamed to `master` when merging.

jewzaam replied:
This is generated, so it will be changed when we merge.

jewzaam replied:
I have reverted `catalog-manifests/` to what it was prior to updates, this one generated version is gone and the channel name reset to what it was, `production`. Subsequent builds will set it based on the branch.

rogbas commented on Makefile:
```diff
@@ -0,0 +1,153 @@
+	# create CatalogSource yaml
+	TEMPLATE=scripts/templates/template_osd-operators.CatalogSource.yaml; \
+	DEST=manifests/osd-operators.CatalogSource.yaml; \
+	$(SED_CMD) $$TEMPLATE > $$DEST
```
Being the pre-req to all the rest, I'd prefix the `CatalogSource` generated yaml with 00 to guarantee it will be the first file under manifests, in case someone (or script) applies all the yamls in the manifests/ folder

jewzaam replied:
Technically it won't matter the order because everything will be reconsiled in the cluster as things are available.  Only requirement is the namespace exist before operatorgroup and subscription are created.  Might be worth putting that in a single file...

jewzaam replied:
It's in a single file now for each operator and I've added a prefix to each.  All the operators have the same prefix as there's no dependency between them and can be installed in any order.

jewzaam committed:
Moved operator checkout to function, updated README on how to add another operator

lisa commented:
I have some feedback for this that is going to have to come after a meeting, so please hold off.

/hold

jewzaam committed:
Moved templates around, cleaned up readme, failure to gen will cleanup catalog-manifests

lisa commented on Dockerfile:
```diff
@@ -0,0 +1,6 @@
+FROM quay.io/openshift/origin-operator-registry:latest
```
We should pin to a specific tag, if possible, and not `latest`.

lisa committed:
Suggested changes for #2

Summary of changes:

Makefile:

* Move list of operators into a Makefile variable `$OPERATORS`
* Silence `isclean` output
* Explictly pass `ALLOW_DIRTY_CECKOUT` to "child" `make` calls (eg `make env`)
* Resolve make bash error in `manifests/operators` target
* Use Makefile variable `$OPERATORS` to loop over and fetch source
* Git source fetcher is a little more friendly:
** Checks to see if the directory exists and git pulls, otherwise clones
** Resolves minor errors when the directory does exist

scripts/gen_operator_csv.py:

* Fix YAML loading error by changing Loader to use `yaml.BaseLoader` instead of `yaml.FullLoader`

Signed-off-by: Lisa Seelye <lseelye@redhat.com>

lisa committed:
Fix syntax error in clean target

Signed-off-by: Lisa Seelye <lseelye@redhat.com>

lisa committed:
Revert `@` before isclean

Signed-off-by: Lisa Seelye <lseelye@redhat.com>

lisa committed:
Loader=BaseLoader results in malformed YAML output

Signed-off-by: Lisa Seelye <lseelye@redhat.com>

lisa committed:
Address Naveen's feedback in checkout-operator.mk

Allow `SKIP_GITREFRESH` to be used to skip doing `git reset` and `git
pull` on the checked out copies of the source code.

Signed-off-by: Lisa Seelye <lseelye@redhat.com>

jewzaam committed:
Update checkout-operator.mk

Co-Authored-By: lisa <18159+lisa@users.noreply.github.com>

jewzaam committed:
Update Makefile

Co-Authored-By: lisa <18159+lisa@users.noreply.github.com>

lisa committed:
Immediate assignment of operator source dir

jewzaam committed:
Update Makefile

Co-Authored-By: lisa <18159+lisa@users.noreply.github.com>

jewzaam committed:
Pin origin-operator-registry to tag v4.0

jewzaam committed:
Update Makefile

Co-Authored-By: lisa <18159+lisa@users.noreply.github.com>

lisa committed:
this var and others are inherited

jewzaam committed:
Merge pull request #1 from lisa/lseelye-feedback-for-2

Suggested changes for #2

jewzaam committed:
Removed updates to the catalog-manifests/, it's back to what is in master branch

lisa reviewed (approved):
/lgtm

jewzaam committed:
Use yaml.safe_load

lisa commented on scripts/gen_operator_csv.py:
```diff
@@ -64,7 +64,7 @@
 
     with open('scripts/templates/csv.yaml', 'r') as stream:
-        csv = yaml.load(stream)
+        csv = yaml.safe_load(stream)
```
See https://security.openstack.org/guidelines/dg_avoid-dangerous-input-parsing-libraries.html

lisa reviewed (approved):
/lgtm
